src/nvim/ops.c

```diff
@@ -2835,8 +2835,8 @@ static void op_yank_reg(oparg_T *oap, bool message, yankreg_T *reg, bool append)
     curr->y_size = j;
     xfree(reg->y_array);
   }
-  if (curwin->w_p_rnu) {
-    redraw_later(curwin, SOME_VALID);  // cursor moved to start
+  if (curwin->w_p_rnu && yanklines > 1) {
+    redraw_later(curwin, VALID);  // cursor moved to start
```

Comment:
This breaks 'cursorline' redrawing only when 'relativenumber' is set. This is strange.

Reply:
Fixed